Spec review fixes (#1112)
riscv/riscv-cheri: src/cheri/riscv-unpriv-integration.adoc

* The _length_ is returned by the <<GCLEN>> instruction.
* The _top_ is returned by the <<GCTOP>> instruction.

> NOTE: The _length_ and _top_ are (XLEN+1)-bit values, so <<GCLEN>> and <<GCTOP>> saturate their result to the maximum XLEN-bit value (2^XLEN^-1) when it does not fit in an XLEN-bit register.

Reviewer: ```suggestion
NOTE: The _length_ and _top_ are (XLEN+1)-bit values, so <<GCLEN>> and <<GCTOP>> saturate their result to the maximum XLEN-bit value (2^XLEN^-1) when it overflows an XLEN-bit register.
```

stored in naturally aligned addresses. {ctag_cap}s must be atomically bound to the
data they protect.

Address arithmetic is modulo 2^XLEN^, so the byte at address 2^XLEN^ - 1 is adjacent to the byte at address zero.
A capability's <<section_cap_representable_check>> is also circular, so address 0 is within the <<section_cap_representable_check>> of a capability where address 2^XLEN^ - 1 is within the bounds.

> However, the decoded top address of a capability is XLEN + 1 bits wide and does *not* wrap.

Reviewer: ```suggestion
However, the decoded top bound address of a capability is XLEN + 1 bits wide and does *not* wrap.
```

Because they are immutable, the jump target cannot be modified, ensuring that control flow can only enter at the specific address encoded in the capability and not elsewhere in a function.

A sealed entry point capability may be passed as the `{cs1}` argument to <<JALR_CHERI>>, and is _unsealed_ on dereference.

> If `{cd}` is not `0`, the link capability written to `{cd}` is also written as a <<sentry_cap>>.

Reviewer: ```suggestion
If `{cd}` is not `{creg}0`, the link capability written to `{cd}` is also written as a <<sentry_cap>>.
```

NOTE: These checks are much less rigorous than parity or ECC protection, and are only
used to detect simple problems with the capability metadata.

A capability whose {ctag} is set can always be assumed to satisfy these integrity rules, because it could only have been produced by a valid derivation.

> Consequently, an integrity check can only change the result of an operation when it is performed on an input capability whose {ctag} is zero.

Reviewer: This is more-or-less a duplicate of the next line - maybe delete the next one?

Reviewer: I'll worry about this later

=== Extended State

> As stated above, all state which can hold addresses is extended from XLEN to YLEN bits.

Reviewer: ```suggestion
As stated above, unless noted otherwise, state which can hold addresses is extended from XLEN to YLEN bits.

NOTE: The privileged `mconfigptr` CSR is the only documented exception to the rule.

```
